packages/ra-ui-materialui/src/input/ResettableTextField.js

```diff
@@ -30,103 +29,107 @@ const styles = createStyles({
+                                [visibleClearButton]:
+                                    clearAlwaysVisible || showClear,
+                            })}
+                            aria-label={translate(
+                                'ra.action.clear_input_value'
+                            )}
+                            title={translate('ra.action.clear_input_value')}
+                            disableRipple
+                            onClick={handleClickClearButton}
+                            onMouseDown={handleMouseDownClearButton}
+                            disabled={disabled}
+                        >
+                            <ClearIcon
+                                className={classNames(clearIcon, {
+                                    [visibleClearIcon]:
                                         clearAlwaysVisible || showClear,
                                 })}
-                                aria-label={translate(
-                                    'ra.action.clear_input_value'
-                                )}
-                                title={translate('ra.action.clear_input_value')}
-                                disableRipple
-                                onClick={this.handleClickClearButton}
-                                onMouseDown={this.handleMouseDownClearButton}
-                                disabled={disabled}
-                            >
-                                <ClearIcon
-                                    className={classNames(clearIcon, {
-                                        [visibleClearIcon]:
-                                            clearAlwaysVisible || showClear,
-                                    })}
-                                />
-                            </IconButton>
-                        </InputAdornment>
-                    ),
-                    ...InputProps,
-                }}
-                disabled={disabled}
-                {...props}
-                onFocus={this.handleFocus}
-                onBlur={this.handleBlur}
-            />
-        );
-    }
+                            />
+                        </IconButton>
+                    </InputAdornment>
+                ),
+                ...InputProps,
+            }}
+            disabled={disabled}
+            {...props}
+            onFocus={handleFocus}
+            onBlur={handleBlur}
+        />
+    );
 }
 
-export default compose(
-    translate,
-    withStyles(styles)
-)(ResettableTextField);
+ResettableTextField.propTypes = {
+    classes: PropTypes.object,
+    clearAlwaysVisible: PropTypes.bool,
+    disabled: PropTypes.bool,
+    InputProps: PropTypes.object,
+    onBlur: PropTypes.func,
+    onChange: PropTypes.func.isRequired,
+    onFocus: PropTypes.func,
+    resettable: PropTypes.bool,
+    translate: PropTypes.func.isRequired,
+    value: PropTypes.any.isRequired,
+};
+
+export default translate(ResettableTextField);
```

Comment:
can you use the `useTranslate` hook to avoid decorating this component with `translate`?